File: landscape-webui/src/lib/docker/network.ts
export class LandscapeDockerNetwork {
  name: string;
  iface_name: string;
  id: string;
  driver?: string;
  containers: Map<string, LandscapeDockerNetworkContainer>;
  options: Map<string, string>;

  constructor(obj: {
    name: string;
    iface_name: string;
    id: string;
    driver?: string;
    containers: Map<string, LandscapeDockerNetworkContainer>;
    options: Map<string, string>;
  }) {
    this.name = obj.name;
    this.iface_name = obj.iface_name;
    this.id = obj.id;
    this.driver = obj.driver;

    let map = new Map<string, LandscapeDockerNetworkContainer>();
    for (const [key, value] of Object.entries(obj?.containers)) {
      map.set(key, new LandscapeDockerNetworkContainer(value));
    }
    this.containers = map;

    let options = new Map<string, string>();
    for (const [key, value] of Object.entries(obj?.containers)) {
      options.set(key, value as string);
    }
    this.options = options;
  }
}

export class LandscapeDockerNetworkContainer {
  name: string;
  mac: string | undefined;
  constructor(obj: { name: string; mac?: string }) {
    this.name = obj.name;
    this.mac = obj.mac;
  }
}

File: landscape-common/src/docker/network.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use ts_rs::TS;

use crate::net::MacAddr;

#[derive(Serialize, Deserialize, Debug, Clone, Default, TS)]
#[ts(export, export_to = "docker.ts")]
pub struct LandscapeDockerNetwork {
    // Name
    pub name: String,
    pub id: String,
    pub driver: Option<String>,
    pub containers: HashMap<String, LandscapeDockerNetworkContainer>,
    pub iface_name: String,
    pub options: HashMap<String, String>,
}

#[derive(Serialize, Deserialize, Debug, Clone, Default, TS)]
#[ts(export, export_to = "docker.ts")]
pub struct LandscapeDockerNetworkContainer {
    pub name: String,
    pub mac: Option<MacAddr>,
}
